WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationDomainAdapter.java

```diff
@@ -0,0 +1,277 @@
+package org.wordpress.android.ui.accounts.signup;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.EditorInfo;
+import android.widget.EditText;
+import android.widget.RadioButton;
+import android.widget.TextView;
+
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.util.helpers.Debouncer;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class SiteCreationDomainAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private static final int VIEW_TYPE_HEADER = 0;
+    private static final int VIEW_TYPE_INPUT = 1;
+    private static final int VIEW_TYPE_ITEM = 2;
+
+    private static final int GET_SUGGESTIONS_INTERVAL_MS = 400;
+
+    public interface OnAdapterListener {
+        void onKeywordsChange(String keywords);
+        void onSelectionChange(String domain);
+    }
+
+    private boolean mIsLoading;
+    private String mInitialKeywords;
+    private List<String> mSuggestions;
+    private OnAdapterListener mOnAdapterListener;
+
+    private String mCarryOverDomain;
+    private String mSelectedDomain;
+    private boolean mNeedExtraLineForSelectedDomain;
```

Comment:
I don't follow the difference between `mCarryOverDomain` and `mSelectedDomain`.  When are they different?  Also, wouldn't `mNeedExtraLineForSelectedDomain` be the same as if `mSelectedDomain` has no value?

Reply:
`Carry-over` and `selected` can be different when the user is on a screen where there is the top-item that got carried over from the previous query and then the user has selected a different item, from the new query. So, on screen, there will be both the carry-over (at the top) and the selected (somewhere down the list).

So, need to keep them in separate vars so when the list refreshes (happens when selecting another list item for instance) the adapter knows what to render and where.

Let me know if that makes sense @theck13 , thanks!